tools/cpp/windows_cc_configure.bzl

```diff
@@ -227,9 +228,38 @@ def find_msvc_tool(repository_ctx, vc_path, tool):
     tool_path = vc_path + "\\bin\\amd64\\" + tool
 
   if not repository_ctx.path(tool_path).exists:
-    auto_configure_fail(tool_path + " not found, please check your VC++ installation.")
+    return None
+
   return tool_path
 
+def _check_vc_installation(repository_ctx, vc_path):
+  """Check if the VC++ installation is valid."""
+  if vc_path == "visual-studio-not-found":
+    tpl(repository_ctx, "vc_path_not_found.bat", {"%{vc_error_message}" : ""})
+    return False
+
+  missing_tools = []
+  if not _find_vcvarsall_bat_script(repository_ctx, vc_path):
+    missing_tools.append("VCVARSALL.BAT")
+
+  for tool in ["cl.exe", "link.exe", "lib.exe", "ml64.exe"]:
+    if not find_msvc_tool(repository_ctx, vc_path, tool):
+      missing_tools.append(tool)
+
+  if missing_tools:
+    tpl(repository_ctx, "vc_path_not_found.bat", {
```

Comment:
Same here: expanding the template is a side-effect, so the method does more than just checking the VC path. Please do this at the call site.

Reply:
Done.